### ISSUE TITLE
Error making handshake

### ISSUE THREAD

**Issue by jdkealy** [ptaoussanis/sente#189]
Hi, 

I built a project around Reagent Template and am trying to get up and running with Sente. 

When I tried using http-kit, I get the following error: 
 No implementation of method: :on-close of protocol: #'org.httpkit.server/Channel found for class: nil

When I tried using immutant: 
[taoensso.sente.server-adapters.immutant      :refer (sente-web-server-adapter)]

I got this error: 
No method in multimethod 'initialize-stream' for dispatch value: :servlet

Any idea what I should try to fix this ?

**Comment by danielcompton** [comment]
Hi there, have you tried using the Sente template to see if you're getting the errors there too? It sounds like something isn't quite configured right.

**Comment by jdkealy** [comment]
By the Sente template, do you mean the cljx app that's included in the Repo ?

**Comment by danielcompton** [comment]
Yep

**Comment by jdkealy** [comment]
thanks i'll try that.

**Comment by jdkealy** [comment]
I got it working with another template here:
https://github.com/enterlab/rente

I'd run with it but it uses the component model. I had a hunch it was the version of figwheel, but the two use the same version "0.5.0-2".

scratching my head...

**Comment by jdkealy** [comment]
I also had a hunch it was "ring-defaults"... Which I removed and still have the error: 

rtest.repl=> java.lang.IllegalArgumentException: No implementation of method: :on-close of protocol: #'org.httpkit.server/Channel found for class: nil
                  core_deftype.clj:554 clojure.core/-cache-protocol-fn
                         server.clj:36 org.httpkit.server/eval13971[fn]
                       http_kit.clj:23 taoensso.sente.server-adapters.http-kit/taoensso.sente.server-adapters.http-kit.HttpKitAsyncNetworkChannelAdapter
                         sente.clj:481 taoensso.sente/make-channel-socket![fn]
                       handler.clj:241 rtest.handler/fn
                          core.clj:130 compojure.core/make-route[fn]
                          core.clj:118 compojure.core/wrap-route-middleware[fn]
                           core.clj:41 compojure.core/if-route[fn]
                           core.clj:27 compojure.core/if-method[fn]
                          core.clj:144 compojure.core/routing[fn]
                         core.clj:2570 clojure.core/some
                          core.clj:144 compojure.core/routing
                       RestFn.java:139 clojure.lang.RestFn.applyTo
                          core.clj:632 clojure.core/apply
                          core.clj:149 compojure.core/routes[fn]


These are my deps
               :dependencies [[org.clojure/clojure "1.7.0"]
                 [ring-server "0.4.0"]
                 [reagent "0.5.1"
                  :exclusions [org.clojure/tools.reader]]
                 [reagent-forms "0.5.13"]
                 [reagent-utils "0.1.7"]
                 [org.clojure/core.async "0.2.374"]
                 [http-kit "2.1.19"]
                 [ring "1.4.0"]
                 [com.datomic/datomic-free "0.9.5130" :exclusions [joda-time]]
                 [clj-redis "0.0.12"]
                 [com.taoensso/sente "1.7.0"]
                 [cheshire "5.5.0"]
                 [org.clojure/clojure-contrib "1.2.0"]
                 [compojure "1.4.0"]
                 [hiccup "1.0.5"]
                 [cljs-ajax "0.3.13"]
                 [fogus/ring-edn "0.3.0"]
                 [clj-time "0.11.0"]
                 [crypto-random "1.2.0"]
                 [environ "1.0.1"]
                 [org.clojure/clojurescript "1.7.170" :scope "provided"]
                 [secretary "1.2.3"]
                 [venantius/accountant "0.1.6" :exclusions [org.clojure/tools.reader]]]

and plugins:

  :plugins [[lein-environ "1.0.1"]
            [lein-cljsbuild "1.1.1"]
            [lein-asset-minifier "0.2.4"
             :exclusions [org.clojure/clojure]]]

**Comment by danielcompton** [comment]
I think your best option is to start form a known good configuration, and slowly add pieces back until you see what is breaking it. There's a lot going on here, and it's pretty hard as an outsider to debug.

**Comment by jdkealy** [comment]
Thanks @danielcompton. I figured :/ . Might you recommend a template to get bootstrapped with fighweel cljs / compure / ring that you know works with Sente?

**Comment by danielcompton** [comment]
Not sure apart from Rente sorry.

**Comment by ptaoussanis** [comment]
Hi John,

Apart from what Daniel's suggested,

> No implementation of method: :on-close of protocol: #'org.httpkit.server/Channel found for class: nil

This error looks like a dependency and/or compilation issue. I'd start by taking a look at your `lein deps :tree` output to look for possible conflicts. Please also make sure to run `lein clean` after any changes to clear out possible stale build artefacts.

Hope that helps, cheers :-)

**Comment by emiguelt** [comment]
Hi All, I'm having the same problem.
Did anybody solve it?
Best regards

**Comment by veikman** [comment]
I’m building an application with Luminus. When I added Sente, I also added Immutant, through a dependency on `luminus-immutant`, so I could use Sente’s Immutant adapter. By mistake, I retained a dependency on `luminus-jetty`, the default HTTP server for Luminus.

I got “No method in multimethod 'initialize-stream' for dispatch value: :servlet” until I removed `luminus-jetty` as a dependency, leaving only one server (Immutant).
